packages/machine/src/protocol/install-virtual-app.ts

```diff
@@ -133,13 +137,31 @@ export const INSTALL_VIRTUAL_APP_PROTOCOL: ProtocolExecutionFlow = {
   ]
 };
 
-function createTarget(
-  signingKeys: string[],
+function createAndAddTarget(
   defaultTimeout: number,
   appInterface: AppInterface,
-  initialState: SolidityABIEncoderV2Struct
-) {
-  return new AppInstance(
+  initialState: SolidityABIEncoderV2Struct,
+  context: Context,
+  initiatingXpub: string,
+  respondingXpub: string,
+  intermediaryXpub: string
+): AppInstance {
+  const key = virtualChannelKey(
+    [initiatingXpub, respondingXpub],
+    intermediaryXpub
+  );
+
+  const sc =
+    context.stateChannelsMap.get(key) ||
+    StateChannel.createEmptyChannel(key, [initiatingXpub, respondingXpub]);
+
+  const appSeqNo = sc.numInstalledApps;
+
+  const signingKeys = [xkeyKthAddress(intermediaryXpub, appSeqNo)].concat(
+    xkeysToSortedKthAddresses([initiatingXpub, respondingXpub], appSeqNo)
+  );
```

Comment:
Why isn't the whole list sorted?

Reply:
https://github.com/counterfactual/monorepo/blob/master/packages/contracts/contracts/mixins/MixinVirtualAppSetState.sol#L99

Reply:
https://github.com/counterfactual/monorepo/blob/master/packages/contracts/contracts/mixins/MixinVirtualAppSetState.sol#L19

Reply:
Got it, can you reference the specs in a comment in the code